src/components/@core/messaging/subscribe-button.tsx

```diff
@@ -0,0 +1,25 @@
+import { Button } from "@chakra-ui/core";
+import { axSaveFCMToken } from "@services/user.service";
+import React from "react";
+
+function SubscribePushButton() {
+  const subscribePush = async () => {
+    const w = window as any;
+    await w.workbox.register();
+    Notification.requestPermission(async (status) => {
+      if (status === "granted") {
+        const token = await w.workbox.messageSW({ command: "getFCMToken" });
+        if (token) {
+          const { success } = await axSaveFCMToken({ token });
+          console.debug(success ? "Token Saved" : "Failed to Save Token");
+        } else {
+          console.debug("Notification Rejected");
+        }
+      }
+    });
+  };
```

Comment:
This is duplicate code can you move it to utils womewhere and use it in both files?
`container/index.tsx` and this one?